### ISSUE TITLE
Too many months

### ISSUE THREAD

**Issue by bondjimbond** [Islandora-Labs/islandora_usage_stats_charts#18]
Looking at the example in #15, and also what's happening in the Collection Stats block here (https://arcabc.ca/islandora/object/kora%3Apublications#), I wonder if the compressed lines on the left have anything to do with an error in setting the number of months to display.

I have Charts set to display six months. But the visible stats show seven, and it's probable that the compressed lines are for many other months further back.

![screen shot 2018-07-24 at 9 32 00 am](https://user-images.githubusercontent.com/6069170/43141689-89068e76-8f24-11e8-86dd-fd4ad7f47b1d.png)

If there's a flaw with rendering the configured number of months, then perhaps fixing it would make those lines go away.

**Comment by mjordan** [comment]
I'll take a look.

**Comment by bondjimbond** [comment]
Also: I'm not actually sure whether those lines are a failed attempt to render more bars, or to render a ridiculously compressed scale with too many numbers on. Either way, should be fixable.

**Comment by mjordan** [comment]
Just horsing around with a better implementation of the code that gets the timestamp from 6 months ago at https://github.com/mjordan/islandora_usage_stats_charts/blob/7.x/includes/blocks.inc#L54:

```php
$months = 6;

$date = strtotime(trim($argv[1]) . ' -' . $months . ' months');
print $months . " months ago: " . date('Y-m-d', $date) . "\n";

$rounded_down = date('Y-m' . '-01', $date);
$rounded_down_timestamp = strtotime($rounded_down);
print $months . " months ago, rounded down to the first of the month: $rounded_down ($rounded_down_timestamp)\n";
```

Running that code, `php php_months_ago.php 2018-01-29`, gets you
```
6 months ago: 2017-07-29
6 months ago, rounded down to the first of the month: 2017-07-01 (1498892400)
```

**Comment by mjordan** [comment]
@bondjimbond the number of months being displayed for me is always the number I specify in my admin settings. In other words, I can't reproduce this problem. I am running the charts module on the latest commit, cf18fa097e937a96f8bde3a18738e3f07bd8cffe.

Can  you do some more testing to determine if the number of months being displayed for you is always one higher than your admin settings for this module indicate? And does this happen on all objects or just some? Keep in mind that stats only go as far back as either 1) the time when you started collecting them using the Islandora Usage Stats module or 2) when the object was added to the repo; the number of months data available for a given object will be displayed at the top of the chart. But these two variables will result in fewer months being displayed, not *more* as you are reporting.

**Comment by mjordan** [comment]
Opps, I correct mysefl: http://digital.lib.sfu.ca/bclhc-1/interview-cathy-walker is showing 7 months worth.

**Comment by mjordan** [comment]
@bondjimbond can you test the issue-18 branch? It should fix the problem of not showing the correct number of bars. We can then move on to the compressed bars problem.

**Comment by mjordan** [comment]
@bondjimbond see second screenshot in #20.

**Comment by bondjimbond** [comment]
issue-18 solves this problem. Y-axis still horribly compressed, but we've now got the right number of months.

**Comment by mjordan** [comment]
OK, thanks, I'll merge that in.

**Comment by mjordan** [comment]
Wrong number of months fixed with 168f77b583c4812e40231274d9a30e34814d4fa9. See #20 for the "compressed lines" issue.
